lib/ash/policy/authorizer/authorizer.ex

```diff
@@ -712,6 +712,25 @@ defmodule Ash.Policy.Authorizer do
     end
   end
 
+  defp do_replace_ref(
+         %{
+           attribute: %struct{name: name},
+           relationship_path: [relationship_path],
+           resource: resource
+         } = ref,
+         %{stack: [{parent, _path, _action} | _]} = acc
+       )
+       when struct in [Ash.Resource.Attribute, Ash.Resource.Aggregate, Ash.Resource.Calculation] do
+    action =
+      Ash.Resource.Info.relationship(parent, relationship_path).read_action ||
+        Ash.Resource.Info.primary_action!(resource, :read)
+
+    {expr, acc} =
+      expression_for_ref(resource, name, action, ref, acc)
```

Comment:
We might also need to update the stack here. But doing it caused an error for me. 

I hit this case when coming from the `%{__function__?: true, ... }` above. Returning the original stack there might be the correct way to do things. What do you think?

Reply:
🤔 I'm pretty sure the way you have it here is the right way :)